### PR TITLE
Clean up error messages for pre-bound PVCs.

### PR DESCRIPTION
**Pull request by jsafrane**
When a PVC explicitly requests specific PV and the PV does not match, we should tell the user what exactly does not match.

From: `Volume's size is smaller than requested or volume's class does not match with claim`

To: `Cannot bind to requested volume "<volume name>": %s`, where `%s` is one of:
- `requested PV is too small`
- `storageClasseNames do not match`
- `incompatible volumeMode`
- `error checking volumeMode: api defaulting for volumeMode failed` (this should not ever happen)

/sig storage
@kubernetes/sig-storage-pr-reviews 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by mtanino** [comment]
@jsafrane 
Can you show what kind of error does user actually see when they use `kubectl describe pvc`?
Can they see volume name, claim name, and exact error msg as you mentioned, right?

**Comment by mtanino** [comment]
I guess PVC name is shown in `where %s` part.

**Comment by jsafrane** [comment]
It's event on PVC so the PVC is implied.

```
$ kubectl describe pvc
Name:          claim-block
Namespace:     default
...
Events:
  Type     Reason          Age   From                         Message
  ----     ------          ----  ----                         -------
  Warning  VolumeMismatch  3s    persistentvolume-controller  Cannot bind to requested volume "block-pv": incompatible volumeMode
```

**Comment by mtanino** [comment]
@jsafrane 
Ah, now I see. lgtm.

**Comment by mtanino** [comment]
cc @screeley44 @erinboyd

**Comment by jsafrane** [comment]
/assign @mtanino 
pretty please :-)

File: pkg/controller/volume/persistentvolume/pv_controller.go [hunk 1]
```diff
@@ -235,22 +235,22 @@ func checkVolumeSatisfyClaim(volume *v1.PersistentVolume, claim *v1.PersistentVo
...
 	volumeSize := volumeQty.Value()
 	if volumeSize < requestedSize {
-		return fmt.Errorf("Storage capacity of volume[%s] requested by claim[%v] is not enough", volume.Name, claimToClaimKey(claim))
+		return fmt.Errorf("requested PV is too small")
 	}
 
 	requestedClass := v1helper.GetPersistentVolumeClaimClass(claim)
 	if v1helper.GetPersistentVolumeClass(volume) != requestedClass {
-		return fmt.Errorf("Class of volume[%s] is not the same as claim[%v]", volume.Name, claimToClaimKey(claim))
+		return fmt.Errorf("storageClasseNames do not match")
```
Comment (mtanino) [inline pkg/controller/volume/persistentvolume/pv_controller.go]: Why does the storage class name take plural?
Should we return `storageClassName does not match`?

Comment (zhangxiaoyu-zidif) [reply pkg/controller/volume/persistentvolume/pv_controller.go]: maybe just like， your password and confirmation password do not match.

Comment (jsafrane) [reply pkg/controller/volume/persistentvolume/pv_controller.go]: Fixed to singular.

Comment (mtanino) [reply pkg/controller/volume/persistentvolume/pv_controller.go]: >> Fixed to singular.

Then, the message is storageClassName `does` not match, right?

**Pull request merged**